kentikapi/client.go

```diff
@@ -132,25 +122,34 @@ func (c *Config) FillDefaults() {
 		c.APIURL = APIURLUS
 	}
 
-	if c.CloudExportHostPort == "" {
-		c.CloudExportHostPort = cloudExportHostPort
-	}
-
-	if c.SyntheticsHostPort == "" {
-		c.SyntheticsHostPort = syntheticsHostPort
-	}
-
 	if c.Timeout == nil {
 		c.Timeout = pointer.ToDuration(defaultTimeout)
 	}
 
 	c.RetryCfg.FillDefaults()
 }
 
-func makeConnForGRPC(c Config, hostPort string) (grpc.ClientConnInterface, error) {
+func parseKentikURL(apiURL *string) (string, bool) {
+	tls := false
+	u, err := url.Parse(*apiURL)
+	if err == nil {
+		*apiURL += "/api/v5"
+		grpc := u.Host
+		if u.Scheme == "https" {
+			tls = true
+			if u.Port() == "" {
+				grpc += ":443"
+			}
+		}
+		return fmt.Sprintf("grpc." + grpc), tls
```

Comment:
Instead of building the URL string manually, please modify url object and then call String() on it.

Reply:
Added comment for API v5 below.. For API v6 using url.String() is not desirable, indeed, because we need only host+port here.
This one is done.